test/test-lex.cpp

```diff
@@ -118,12 +118,109 @@ TEST(test_lex, lex_binary_numbers) {
   check_single_token(u8"0B010101010101010", token_type::number);
 }
 
+// fails
```

Comment:
I added this test because I found a potential for a bug while reading the binary lexer.  Didn't fix it, just wrote the test.  Maybe I should disable this because the test fails